test/powershell/Modules/Microsoft.PowerShell.Diagnostics/CounterTestHelperFunctions.ps1

```diff
@@ -0,0 +1,310 @@
+
+                if (result == 0)
+                    rv = Marshal.PtrToStringUni(pathPtr);
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(pathPtr);
+            }
+
+            return rv;
+        }
+    }
+"@
+
+Add-Type -TypeDefinition $helperSource
+
+# Given a performance-counter ID, look up and return the localized name
+#
+# This function came from an article in PowerShellMagazine by Tobias Weltner
+# http://www.powershellmagazine.com/2013/07/19/querying-performance-counters-from-powershell/
+Function Get-PerformanceCounterLocalName
+{
+    param
+    (
+        [UInt32]
+        $ID,
+
+        $ComputerName = $env:COMPUTERNAME
+    )
+
+    $code = '[DllImport("pdh.dll", SetLastError=true, CharSet=CharSet.Unicode)] public static extern UInt32 PdhLookupPerfNameByIndex(string szMachineName, uint dwNameIndex, System.Text.StringBuilder szNameBuffer, ref uint pcchNameBufferSize);'
+
+    $Buffer = New-Object System.Text.StringBuilder(1024)
+    [UInt32]$BufferSize = $Buffer.Capacity
+
+    $t = Add-Type -MemberDefinition $code -PassThru -Name PerfCounter -Namespace Utility
+    $rv = $t::PdhLookupPerfNameByIndex($ComputerName, $id, $Buffer, [Ref]$BufferSize)
+
+    if ($rv -eq 0)
+    {
+        $Buffer.ToString().Substring(0, $BufferSize-1)
+    }
+    else
+    {
+        Throw 'Get-PerformanceCounterLocalName : Unable to retrieve localized name. Check computer name and performance counter ID.'
+    }
+}
+
+# Given a performance-counter name, look up and return the counter ID
+#
+# This function is a slightly modified version of the function originally published on
+# PowerShellMagazine by Tobias Weltner
+# http://www.powershellmagazine.com/2013/07/19/querying-performance-counters-from-powershell/
+function Get-PerformanceCounterID
+{
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        $Name
+    )
+
+    if ($script:perfhash -eq $null)
+    {
+        $script:perfhash = @{}
+    }
+
+    if (-not $script:perfHash.$Name)
+    {
+        $key = 'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Perflib\CurrentLanguage'
+        $counters = (Get-ItemProperty -Path $key -Name Counter).Counter
+        $all = $counters.Count
+
+        for ($i = 0; $i -lt $all; $i += 2)
+        {
+            if ([string]::Compare($counters[$i + 1], $Name, $true) -eq 0)
+            {
+                $script:perfHash.$($counters[$i + 1]) = $counters[$i]
+            }
+        }
+    }
+
+    $script:perfHash.$Name
+}
+
+# Translate a counter path from "English" to a localized counter path
+function TranslateCounterPath($path)
+{
+    $counterHelper = New-Object -TypeName "TestCounterHelper"
+    $rv = $path;
+
+    # crack the counter path
+    $parts = $counterHelper.ParseCounterPath($path);
+
+    if ($parts)
+    {
+        $objectName = $null
+        $counterName = $null
+        $id = Get-PerformanceCounterID $parts.ObjectName
+        if ($id -ne $null)
```

Comment:
I had been thinking of $id as a number (with the possibility of 0 as a valid ID), rather than a string containing digits, which is what it is. Simply testing `if ($id)` is more appropriate. Will fix.